src/MonoTorrent.Client/MonoTorrent.Connections/HttpPeerConnection.cs

```diff
@@ -357,23 +357,41 @@ void CreateWebRequests (BlockInfo start, BlockInfo end)
                     break;
 
                 // We want data from a later file
-                if (startOffset >= file.Length) {
-                    startOffset -= file.Length;
-                    endOffset -= file.Length;
+                if (startOffset >= file.Length + file.Padding) {
+                    startOffset -= file.Length + file.Padding;
+                    endOffset -= file.Length + file.Padding;
                 }
                 // We want data from the end of the current file and from the next few files
-                else if (endOffset >= file.Length) {
-                    WebRequests.Enqueue ((u, startOffset, file.Length - startOffset));
-                    startOffset = 0;
-                    endOffset -= file.Length;
+                else if (endOffset >= (file.Length + file.Padding)) {
+                    // Maybe we need data from the actaul file?
+                    if (file.Length - startOffset < 0) {
+                        WebRequests.Enqueue ((u, startOffset, file.Length - startOffset));
```

Comment:
If `file.Length - startOffset < 0`, I think we end up requesting a negative number of bytes here. Is that right?

Reply:
Ah, I had added a bunch of "throw" calls for all the negative ranges so I could catch the issue at instantiation time, not later on when the web request was used.

This branch is very much a WIP and doesn't pass tests yet - though that's still a good catch! There's a good chance I wouldn't notice that immediately and would waste a bunch of time trying to get the tests green 😆